app/internal/import_file.py

```diff
@@ -0,0 +1,178 @@
+import datetime
+import os
+import re
+from typing import Any, Dict, List, Tuple, Union
+
+from icalendar import Calendar
+
+from app.database.models import Event
+from app.database.database import SessionLocal
+
+
+NUM_OF_VALUES = 3  # Event contains head, content and date.
+MAX_FILE_SIZE_MB = 5  # 5MB
+VALID_FILE_EXTENSION = (".txt", ".csv", ".ics")  # Can import only these files.
+VALID_YEARS = 20  # Events must be within 20 years range from the current year.
+EVENT_HEADER_NOT_EMPTY = 1  # 1- for not empty, 0- for empty.
+EVENT_HEADER_LIMIT = 50  # Max characters for event header.
+EVENT_CONTENT_LIMIT = 500  # Max characters for event characters.
+MAX_EVENTS_START_DATE = 10  # Max Events with the same start date.
+
+
+def check_file_size(file: str, max_size: int = MAX_FILE_SIZE_MB) -> bool:
+    file_size = os.stat(file).st_size / 1048576  # convert bytes to MB.
+    return file_size <= max_size
+
+
+def check_file_extension(file: str,
+                         extension: Union[str, Tuple[str, ...]]
+                         = VALID_FILE_EXTENSION) -> bool:
+    return file.lower().endswith(extension)
+
+
+def is_file_exist(file: str) -> bool:
+    try:
+        with open(file, "r"):
+            pass
+        return True
+    except (FileNotFoundError, OSError):
+        return False
+
+
+def check_date_in_range(date1: Union[str, datetime.datetime],
+                        valid_dates: int = VALID_YEARS) -> bool:
+    """
+    check if date is valid and in the range according to the rule we have set
+    """
+    now_year = datetime.datetime.now().year
+    if isinstance(date1, str):
+        try:
+            check_date = datetime.datetime.strptime(date1, "%m-%d-%Y")
+        except ValueError:
+            return False
+    else:
+        check_date = date1
+    if check_date.year > now_year + valid_dates or \
+       check_date.year < now_year - valid_dates:
+        return False
+    return True
+
+
+def check_validity_of_txt(row: str) -> bool:
+    """Check if the row contains valid data"""
+    get_values = re.findall(r"^(\w{" + str(EVENT_HEADER_NOT_EMPTY) + "," +
+                            str(EVENT_HEADER_LIMIT) + r"})\,\s(\w{0," +
+                            str(EVENT_CONTENT_LIMIT) +
+                            r"})\,\s(\d{2}\-\d{2}\-\d{4})$", row)
+    if get_values:
+        if len(get_values[0]) == NUM_OF_VALUES:
+            return True
+    return False
```

Comment:
Since you aren't returning the result of the findall, you can use re.search() instead and then just do
```
if get_values:
    return True
return False
```

Regarding the regex itself, commas (",") and dashes ("-") don't need to be escaped.